File: solutions/2108.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.replace(/\r/g, "").split("\n").filter(l => l.length > 0);
lines.pop();

var biggest = "", res = "";
lines.forEach(line => {
  let flag = 0;
  line = line.split(" ");
  line.forEach(word => {
    if (flag)
      res += "-";
    res += word.length;
    if (word.length >= biggest.length)
      biggest = word;
    flag = 1;
  });
  res += "\n";
});

console.log(res);
console.log("The biggest word:",biggest);

File: solutions/2108.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    string s;
    string biggest = "", res = "";
    while (getline(cin, s) && s != "0") {
        
        int flag = 0;
        istringstream ss(s);
        string word;
        vector<string> words;
        while (ss >> word) words.push_back(word);

        for (auto w : words) {
            if (flag) res += "-";
            res += to_string(w.size());
            if (w.size() >= biggest.size()) biggest = word;
            flag = 1;
        }
        res += "\n";
    }
    cout << res << "\n";
    cout << "The biggest word: " << biggest << "\n";

    return 0;
}
